src/recipe/recipe.controller.ts

```diff
@@ -141,4 +148,38 @@ export class RecipeController {
   ): Promise<void> {
     await this.recipeService.deleteRecipe(recipeId, userId);
   }
+
+  @UseGuards(JwtAuthGuard)
+  @ApiOperation({ summary: 'Upload image(s) of recipe' })
+  @ApiBearerAuth()
+  @ApiNotFoundResponse({ description: 'Recipe does not exist' })
+  @ApiUnauthorizedResponse({ description: 'User is not authenticated' })
+  @ApiUnprocessableEntityResponse({ description: 'Wrong image format' })
+  @ApiParam({
+    name: 'id',
+    description: 'Positive integer (≥1) to upload image(s) to specified recipe',
+  })
+  @Post('upload/:id')
+  @UseInterceptors(FilesInterceptor('file'))
+  async uploadFile(
+    @UserId() userId: number,
+    @Param('id', ParseIntPipe) recipeId: number,
+    @UploadedFiles(
+      new ParseFilePipeBuilder()
+        .addMaxSizeValidator({
+          maxSize: 1024 * 1024 * 4,
```

Comment:
this could be exported to const with a name that tells us what unit it is, e.g. `const MAX_IMAGE_SIZE_IN_MB = 1024 * 1024 * 4`